app/src/lib/stores/app-store.ts

```diff
@@ -1728,6 +1729,52 @@ export class AppStore extends TypedBaseStore<IAppState> {
     }
   }
 
+  /**
+   * Cleanup any related UI related to conflicts if still in use.
+   */
+  private clearConflictsFlowVisuals() {
```

Comment:
I feel like we could make this whole method a lot cleaner if we added an optional `popupType` and `bannerType` arguments to the `closePopup` and `clearBanner` methods (similar to what we do in `closeFoldout` except not prescriptive). So something like

```ts
public async _closePopup(popupType?: PopupType): Promise<void> {
  const currentPopup = this.currentPopup

  if (currentPopup == null) {
    return
  }

  if (popupType !== undefined && currentPopup.type !== popupType) {
    return
  }
```

That way we should just be able to list all the popupTypes we want to close here and if any are open they'll get closed

```
this._closePopup(PopupType.MergeConflicts)
this._closePopup(PopupType.AbortMerge)
this._closePopup(PopupType.RebaseFlow)

this._clearBanner(PopupType.MergeConflictsFound)
this._clearBanner(BannerType.RebaseConflictsFound)
```

Reply:
👍 I'm into it. Thoughts @outofambit?

Reply:
yeah, i'm into it!